File: atomic-x/android/src/main/kotlin/io/trtc/tuikit/atomicx/filepicker/FilePickerPlugin.kt
package io.trtc.tuikit.atomicx.filepicker

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodChannel

class FilePickerPlugin : FlutterPlugin {
    private var methodChannel: MethodChannel? = null
    private var handler: FilePickerHandler? = null

    override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        methodChannel = MethodChannel(binding.binaryMessenger, "atomic_x/file_picker")

        handler = FilePickerHandler(
            context = binding.applicationContext,
            methodChannel = methodChannel!!
        )

        methodChannel?.setMethodCallHandler(handler)
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        handler?.dispose()
        handler = null
        methodChannel?.setMethodCallHandler(null)
        methodChannel = null
    }
}

File: atomic-x/ios/Classes/FilePicker/FilePickerPlugin.swift
import Flutter
import UIKit

public class FilePickerPlugin: NSObject {
    private var methodChannel: FlutterMethodChannel?
    private var handler: FilePickerHandler?
    
    init(registrar: FlutterPluginRegistrar) {
        super.init()
        
        methodChannel = FlutterMethodChannel(
            name: "atomic_x/file_picker",
            binaryMessenger: registrar.messenger()
        )
        
        handler = FilePickerHandler(methodChannel: methodChannel!)
        
        methodChannel?.setMethodCallHandler { [weak self] (call, result) in
            self?.handler?.handle(call, result: result)
        }
    }
    
    public func dispose() {
        handler?.dispose()
        handler = nil
        methodChannel?.setMethodCallHandler(nil)
        methodChannel = nil
    }
}
